### PR TITLE
Feature/manage additional emails for websites

### PR DESCRIPTION
**Pull request by ryaaoka**
[Ticket](https://dbmedialab.atlassian.net/browse/RC-294)

Added frontend for ability to choose additional emails receivers:
-escalation emails
-fallback email

### CODE REVIEW

File: src/admin/services/Validation.ts [hunk 1]
```diff
@@ -62,7 +62,19 @@ const additionalRules : IValidationRules = {
...
+		error: 'User mail should be valid email address!',
+	},
+
+	escalation: {
+		type: 'string',
+		exec: customValidations.isEmail,
+		error: 'User mail should be valid email address!',
+	},
+
+	fallback: {
```
Comment (NatalyMac) [inline src/admin/services/Validation.ts]: is it possible to make one rule  - "email"  and do not repeat the same code three times?

File: src/admin/components/website/modalParts/emailOverride/WebsiteOverrideSettings.tsx [hunk 2]
```diff
@@ -17,58 +17,61 @@
...
-		const {value, touched} = this.state;
-		return this.props.checkValidation(value, touched);
+		return this.checkValidation(this.props.controlPropName);
 	}
 
 	onDelete (index: number) {
 		const {list, listPropName} = this.props;
 		if (index >= 0) {
-			const overrideList = list;
+			const overrideList = Immutable(list).asMutable();
```
Comment (liordoe) [inline src/admin/components/website/modalParts/emailOverride/WebsiteOverrideSettings.tsx]: You should work with this as Immutable object - try [this](https://facebook.github.io/immutable-js/docs/#/List/splice), for example

File: src/admin/components/modal/WebsiteModalComponent.tsx [hunk 3]
```diff
@@ -27,7 +27,12 @@ import WebsiteHosts from 'admin/components/website/modalParts/WebsiteHosts';
 import WebsiteEditors from 'admin/components/website/modalParts/WebsiteEditors';
 import WebsiteLayout from 'admin/components/website/modalParts/WebsiteLayout';
 import {WebsiteLayoutProps} from 'admin/types/Website';
-import WebsiteFeedbackEmailOverride from '../website/modalParts/WebsiteFeedbackEmailOverride';
+import WebsiteFeedbackEmailOverride from
+'../website/modalParts/emailOverride/WebsiteFeedbackEmailOverride';
+import WebsiteEscalationEmailOverride from
+'../website/modalParts/emailOverride/WebsiteEscalationEmailOverride';
+import WebsiteFallbackFeedbackEmail from
+'../website/modalParts/emailOverride/WebsiteFallbackFeedbackEmail';
```
Comment (NatalyMac) [inline src/admin/components/modal/WebsiteModalComponent.tsx]: do not remember, it looks like someone told about absolute paths in the import

File: src/admin/components/website/modalParts/emailOverride/WebsiteOverrideSettings.tsx [hunk 4]
```diff
@@ -17,58 +17,61 @@
...
-	checkValidation: (value: string, touched: boolean) => string;
+	isControllable?: boolean;
 	controlPropName?: string;
 	controlPropValue?: boolean;
+	label: JSX.Element | string;
+	ID: string;
+	value: string;
+	touched: boolean;
+	onSubmit: (data: any) => void;
+	onChange: (value:string, touched:boolean) => void;
```
Comment (NatalyMac) [inline src/admin/components/website/modalParts/emailOverride/WebsiteOverrideSettings.tsx]: is it used in this component? may be I missed

File: src/admin/components/website/modalParts/emailOverride/WebsiteBaseEmailOverride.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		super(props);
+		this.state = {
+			value: '',
+			touched: false,
+		};
+	}
+
+	onChange(value:string, touched:boolean) {
+		this.setState({
+			value,
```
Comment (NatalyMac) [inline src/admin/components/website/modalParts/emailOverride/WebsiteBaseEmailOverride.tsx]: really do not understand if it works and where we use this method

File: src/admin/components/website/modalParts/emailOverride/WebsiteOverrideSettings.tsx (hunk 4 above)
Comment (ryaaoka) [reply src/admin/components/website/modalParts/emailOverride/WebsiteOverrideSettings.tsx]: 95 line
https://github.com/dbmedialab/reader-critics/pull/771/files/274414e92804c7ba742c3eb0621c8bb1ac7ffa4e#diff-6f649265f4a0439ed794954bdc7f8b3fR92

**Pull request merged**